features/topology-map/plugins/org.opennms.features.topology.plugins.topo.pathoutage/src/test/java/org/opennms/features/topology/plugins/topo/pathoutage/TestNodeGenerator.java

```diff
@@ -0,0 +1,147 @@
+
+import com.google.common.collect.Lists;
+
+/**
+ * This class with is responsible for generation of test {@link OnmsNode} objects
+ */
+class TestNodeGenerator {
+	/**
+	 * This method generates 9 {@link OnmsNode} objects for testing of
+	 * {@link PathOutageProvider} and {@link PathOutageStatusProvider}
+	 * @param location {@link OnmsMonitoringLocation} or null
+	 * @return Resulting map - values are nodes' levels in the path outage hierarchy
+	 */
+	public static Map<OnmsNode, Integer> generateNodes(OnmsMonitoringLocation location) {
+		Map<OnmsNode, Integer> nodesMap = new HashMap<>();
+
+		OnmsNode node0 = new OnmsNode();
+		node0.setId(1);
+		node0.setLabel("Node-0");
+		if (location != null) {
+			node0.setLocation(location);
+		}
+		nodesMap.put(node0, 0);
+
+		OnmsNode node0_0 = new OnmsNode();
+		node0_0.setId(2);
+		node0_0.setLabel("Node-0-0");
+		node0_0.setParent(node0);
+		if (location != null) {
+			node0_0.setLocation(location);
+		}
+		nodesMap.put(node0_0, 1);
+
+		OnmsNode node0_1 = new OnmsNode();
+		node0_1.setId(3);
+		node0_1.setLabel("Node-0-1");
+		node0_1.setParent(node0);
+		if (location != null) {
+			node0_1.setLocation(location);
+		}
+		nodesMap.put(node0_1, 1);
+
+		OnmsNode node0_1_0 = new OnmsNode();
+		node0_1_0.setId(4);
+		node0_1_0.setLabel("Node-0-1-0");
+		node0_1_0.setParent(node0_1);
+		if (location != null) {
+			node0_1_0.setLocation(location);
+		}
+		nodesMap.put(node0_1_0, 2);
+
+		OnmsNode node0_1_1 = new OnmsNode();
+		node0_1_1.setId(5);
+		node0_1_1.setLabel("Node-0-1-1");
+		node0_1_1.setParent(node0_1);
+		if (location != null) {
+			node0_1_1.setLocation(location);
+		}
+		nodesMap.put(node0_1_1, 2);
+
+		OnmsNode node1 = new OnmsNode();
+		node1.setId(6);
+		node1.setLabel("Node-1");
+		if (location != null) {
+			node1.setLocation(location);
+		}
+		nodesMap.put(node1, 0);
+
+		OnmsNode node2 = new OnmsNode();
+		node2.setId(7);
+		node2.setLabel("Node-2");
+		if (location != null) {
+			node2.setLocation(location);
+		}
+		nodesMap.put(node2, 0);
+
+		OnmsNode node2_0 = new OnmsNode();
+		node2_0.setId(8);
+		node2_0.setLabel("Node-2-0");
+		node2_0.setParent(node2);
+		if (location != null) {
+			node2_0.setLocation(location);
+		}
+		nodesMap.put(node2_0, 1);
+
+		OnmsNode node2_1 = new OnmsNode();
+		node2_1.setId(9);
+		node2_1.setLabel("Node-3");
+		node2_1.setParent(node2_0);
+		if (location != null) {
+			node2_1.setLocation(location);
+		}
+		nodesMap.put(node2_1, 2);
+
+		return nodesMap;
+	}
+
+	/**
+	 * This method provides {@link PathOutageProviderTest} with information about the nodes that can be used for
+	 * hierarchy modification test. It returns the following 4 indices:
+	 * 0: A node that will get a child - future parent node
+	 * 1: Child of a first node
+	 * 2: Child of a second node
+	 * 3: Child of a 3rd node
+	 * 3: Index of an unused node (to add a new one in test)
+	 * @return
+	 */
+	public static List<Integer> getNodesForHierarchyUpdate() {
```

Comment:
This is only used once and would be better placed in the test which actually uses this.